### DISCUSSION TITLE
Filter alerts based on event type

### DISCUSSION THREAD

**Discussion by sgtwtf** [thinkst/opencanary#266] [Feature Requests]
I would love this even more if i could make the email alerts for startup events go to a different email than if an event was triggered.

Autorun prompt during setup or firstrun would be very useful.

**Comment by jayjb** [comment]
Hi @sgtwtf,

Ye its a known issue that we need to sort. We handle it on the commercial side by filtering logtypes (i.e. if log is from http about a login, then alert rather than alert on all logs). 

Ill add it to our update list (otherwise a PR is always welcome!)

**Comment by sgtwtf** [comment]
PR? Is that product request? Is there a place I can submit that for you?

**Comment by jayjb** [comment]
Hi @sgtwtf,

Ah sorry I shouldve explained it. I meant a "Pull Request". So all the code Opencanary uses is in this repository. If you find that you have some time and would like to contribute to this project, you can clone the Opencanary repository, make the changes that you would find useful, and submit them to me as a PR (Pull Request). 

Let me know if this is something you would be keen to try.

**Comment by mgabor3141** [comment]
Logtype based filtering has since been added.

Add the following to your root config object to filter all startup related messages from the logs and only log actual alerts. This filters these messages from the terminal logs as well unfortunately, but that's still better than spamming emails (for example) on every startup.

```json
    "logtype.ignorelist": [ 1000, 1001, 1002, 1003, 1004, 1005, 1006 ],
```

Here's the relevant source code where the logtypes are defined: https://github.com/thinkst/opencanary/blob/master/opencanary/logger.py#L56

Writing this here as this info was rather hard to find, and someone could stumble onto this thread looking for this.

**Reply by emperortomato** [reply]
Thanks for posting this solution -- I spent a while trying to figure out if it was possible to filter out the startup alerts before stumbling upon this thread. Excluding event 1001 did the trick.

If any opencanary devs see this, it would be cool if the [opencanary documentation](https://docs.opencanary.org/en/latest/starting/configuration.html) could be updated to include the `logtype.ignorelist` config directive.
